crates/ruma-common/src/thirdparty.rs

```diff
@@ -255,6 +259,27 @@ pub struct ThirdPartyIdentifier {
     pub added_at: MilliSecondsSinceUnixEpoch,
 }
 
+impl Borrow<str> for ThirdPartyIdentifier {
+    fn borrow(&self) -> &str {
+        &self.address
+    }
+}
```

Comment:
We can implement `AsRef<str>` (not sure why we would) but not this unless we change the other impls to only consider the address field. Since Kévin had some good reasons to oppose that, I think we should just not include this impl at all.